File: backend/catalog-service/src/main/java/com/tripify/catalog_service/dto/RoomTypeDTO.java
package com.tripify.catalog_service.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RoomTypeDTO {
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
    private Integer totalRooms;
    private Integer maxOccupancy;
    private List<String> benefits;
    private List<String> imageUrls;
}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/RoomTypeDto.kt
package com.tripify.tripify_android.data.model

data class RoomTypeDto(
    val id: Int,
    val name: String,
    val description: String?,
    val price: Double,
    val totalRooms: Int,
    val maxOccupancy: Int?,
    val benefits: List<String>?,
    val imageUrls: List<String>?
)
